### ISSUE TITLE
DNS Crash

### ISSUE THREAD

**Issue by C-Sto** [BishopFox/sliver#225]
**Describe the bug**
I didn't get around to fixing it when I said I would, sorry!

tl;dr it appears the dns handler isn't checking for a nil where one could be. I'm not sure how there can be a nil here, but there was. Possibly due to the filesystem filling up, and restarting the server after a session was established, resulting in the server forgetting about it, and the agent insisting it exists 😬 .

**To Reproduce**
I've been unable to reproduce easily, but the situation described above *should* result in the same crash.

**Expected behavior**
No crash pls

**Screenshots**
```
goroutine 97 [running]:                                                                              
github.com/bishopfox/sliver/server/c2.dnsSessionPoll(0xc00604d500, 0x14, 0xc006c38e10, 0x3, 0x3, 0x2,
 0xc006c38e10, 0x3, 0x3, 0xc0000841b0)                                                               
        github.com/bishopfox/sliver/server/c2/udp-dns.go:640 +0x107                                  
github.com/bishopfox/sliver/server/c2.handleTXT(0xc00604d500, 0x14, 0xc0000ff230, 0x18, 0xc00679c510,
 0x1)                                                                                                
        github.com/bishopfox/sliver/server/c2/udp-dns.go:319 +0xcd6                                  
github.com/bishopfox/sliver/server/c2.handleC2(0xc00604d500, 0x14, 0xc00679c510, 0x23)               
        github.com/bishopfox/sliver/server/c2/udp-dns.go:189 +0x17a                                  
github.com/bishopfox/sliver/server/c2.handleDNSRequest(0xc006770950, 0x1, 0x1, 0x1, 0xae2fb00, 0xc006
c1c9c0, 0xc00679c510)                                                                                
        github.com/bishopfox/sliver/server/c2/udp-dns.go:154 +0x1b9                                  
github.com/bishopfox/sliver/server/c2.StartDNSListener.func1(0xae2fb00, 0xc006c1c9c0, 0xc00679c510)  
        github.com/bishopfox/sliver/server/c2/udp-dns.go:131 +0xcb                                   
github.com/miekg/dns.HandlerFunc.ServeDNS(0xc00677eb10, 0xae2fb00, 0xc006c1c9c0, 0xc00679c510)       
        github.com/miekg/dns@v1.1.29/server.go:37 +0x44                                              
github.com/miekg/dns.(*ServeMux).ServeDNS(0xc005fca600, 0xae2fb00, 0xc006c1c9c0, 0xc00679c510)       
        github.com/miekg/dns@v1.1.29/serve_mux.go:103 +0x5d                                          
github.com/miekg/dns.(*Server).serveDNS(0xc006c945a0, 0xc0067e2400, 0x49, 0x200, 0xc006c1c9c0)       
        github.com/miekg/dns@v1.1.29/server.go:609 +0x2e0                                            
github.com/miekg/dns.(*Server).serveUDPPacket(0xc006c945a0, 0xc00673ede0, 0xc0067e2400, 0x49, 0x200, 
0xc00000ee68, 0xc00009b8e0)                                                                          
        github.com/miekg/dns@v1.1.29/server.go:549 +0xb2                                             
created by github.com/miekg/dns.(*Server).serveUDP                                                   
        github.com/miekg/dns@v1.1.29/server.go:479 +0x28a
```

**Additional context**
It appears that this is the function in question:
https://github.com/BishopFox/sliver/blob/e3902dab825ac7d039099a784bbe6a706aa17721/server/c2/udp-dns.go#L626-L648

I would have expected a crash at 633 if no session existed, but apparently that didn't happen.

**Comment by C-Sto** [comment]
I've just realised that's likely the wrong stack trace, and this is a bit of a goose-chase. I'll have to track down/work out how to reproduce properly. Sorry - moving fast and breaking things etc.

**Comment by rkervella** [comment]
Fixed in #403
